pkg/issuer/acme/dns/rfc2136/tsig.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package rfc2136 implements a DNS provider for solving the DNS-01 challenge
+// using the rfc2136 dynamic update.
+
+package rfc2136
+
+import (
+	"crypto/hmac"
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"hash"
+
+	"github.com/miekg/dns"
+)
+
+// tsigHMACProvider is our implementation of github.com/miekg/dns.TsigProvider interface
+// that also supports HMACMD5 as a TSIG algorithm, which is no longer supported
+// by the default TsigProvider implementation in github.com/miekg/dns.
+// For context see the discussion on https://github.com/cert-manager/cert-manager/pull/4942
+type tsigHMACProvider string
+
+var _ dns.TsigProvider = tsigHMACProvider("")
+
+// Generate is a largely copied from
+// github.com/miekg/dns.tsigHMACProvider.Generate except for supporting HMACMD5
```

Comment:
Maybe link to https://github.com/miekg/dns/blob/d70eb7b9e19e2d91e54e655209bdb262c735e6bf/tsig.go#L37-L60 so that it's easy to know exactly what we copied.